core/store/orm/config.go

```diff
@@ -701,7 +716,11 @@ func (c Config) EthNonceAutoSync() bool {
 
 // EthGasLimitDefault sets the default gas limit for outgoing transactions.
 func (c Config) EthGasLimitDefault() uint64 {
-	return c.getWithFallback("EthGasLimitDefault", parseUint64).(uint64)
+	ethGasLimitDefault := chainSpecificConfig(c).EthGasLimitDefault
+	if c.viper.IsSet(EnvVarName("EthGasLimitDefault")) || ethGasLimitDefault == 0 {
+		return c.getWithFallback("EthGasLimitDefault", parseUint64).(uint64)
```

Comment:
getWithFallback is pointless here because if there is a default set, it will always enter this path